treescript/mercurial.py

```diff
@@ -0,0 +1,196 @@
+"""Treescript mercurial functions."""
+# import asyncio
+# from asyncio.subprocess import PIPE, STDOUT
+# import functools
+# import hashlib
+# import json
+import logging
+import os
+import sys
+# from shutil import copyfile
+# import traceback
+# from collections import namedtuple
+
+from treescript.utils import execute_subprocess
+from treescript.exceptions import FailedSubprocess
+from treescript.task import get_source_repo, get_tag_info
+
+# https://www.mercurial-scm.org/repo/hg/file/tip/tests/run-tests.py#l1040
+# For environment vars.
+
+HGRCPATH = os.path.join(os.path.dirname(__file__), 'data', 'hgrc')
+ROBUSTCHECKOUT_PATH = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), 'py2', 'robustcheckout.py')
+)
+TAG_MSG = "No bug - Tagging {revision} with {tags} a=release CLOSED TREE"
+
+log = logging.getLogger(__name__)
+
+
+def build_hg_command(context, *args):
+    """Generate a mercurial command to run.
+
+    See-Also `build_hg_environment`
```

Comment:
This may need to be double-backticks if you send this to RTD.

Reply:
good catch, thanks